x/oracle/keeper/keeper.go

```diff
@@ -43,6 +44,27 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
+func (k Keeper) GetProphecies(ctx sdk.Context) []types.Prophecy {
+	var prophecies []types.Prophecy
+	store := ctx.KVStore(k.storeKey)
+	iter := store.Iterator(nil, nil)
+	for ; iter.Valid(); iter.Next() {
+		key := iter.Key()
+		var dbProphecy types.DBProphecy
+		if bytes.Compare(key, types.AdminAccountPrefix) != 0 &&
```

Comment:
we can use the KVStorePrefixIterator to get all prophecies. current implemenation only works if there are just three prefix for oracle module.

Reply:
@juniuszhou the prophecies haven't been set with any prefix.

Reply:
@juniuszhou - Good point. Yes, we can. Do you have an exact implementation in mind?